### DISCUSSION TITLE
TypeError: 'module' object is not callable

### DISCUSSION THREAD

**Discussion by chikiuso** [coqui-ai/TTS#1681] [General]
Hello, I install the TTS from source and got the TypeError: 'module' object is not callable by running a training script :

Traceback (most recent call last):
  File "jack.py", line 43, in <module>
    train_samples, eval_samples = load_tts_samples(
  File "/root/TTS/TTS/tts/datasets/__init__.py", line 111, in load_tts_samples
    meta_data_train = formatter(root_path, meta_file_train, ignored_speakers=ignored_speakers)
TypeError: 'module' object is not callable

May I have your help? thanks.

**Comment by p0p4k** [comment]
Hello, paste your entire training script. Thanks.

**Reply by chikiuso** [reply]
import os

# BaseDatasetConfig: defines name, formatter and path of the dataset.
from TTS.tts.configs.shared_configs import BaseDatasetConfig

output_path = "./"
if not os.path.exists(output_path):
    os.makedirs(output_path)

dataset_config = BaseDatasetConfig(
    name="dataset", meta_file_train="metadata.csv", path=os.path.join(output_path, "dataset/")
)

# GlowTTSConfig: all model related values for training, validating and testing.
from TTS.tts.configs.glow_tts_config import GlowTTSConfig
config = GlowTTSConfig(
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=100,
    text_cleaner="phoneme_cleaners",
    use_phonemes=True,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=25,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    save_step=1000,
)

from TTS.utils.audio import AudioProcessor
ap = AudioProcessor.init_from_config(config)

from TTS.tts.utils.text.tokenizer import TTSTokenizer
tokenizer, config = TTSTokenizer.init_from_config(config)

from TTS.tts.datasets import load_tts_samples
train_samples, eval_samples = load_tts_samples(
    dataset_config,
    eval_split=True,
    eval_split_max_size=config.eval_split_max_size,
    eval_split_size=config.eval_split_size,
)

from TTS.tts.models.glow_tts import GlowTTS
model = GlowTTS(config, ap, tokenizer, speaker_manager=None)

from trainer import Trainer, TrainerArgs
trainer = Trainer(
    TrainerArgs(), config, output_path, model=model, train_samples=train_samples, eval_samples=eval_samples
)

trainer.fit()

**Comment by p0p4k** [comment]
I believe the error is in the `name="dataset"` . 
The name parameter must correspond to one of the formatters in the [formatters.py](TTS/tts/datasets/formatters.py) file. If you have your own dataset and want to use a custom formatter, you can make a new function and use that function's name in the code. 
```
dataset_config = BaseDatasetConfig(
name="dataset", meta_file_train="metadata.csv", path=os.path.join(output_path, "dataset/")
)
```
